redis.js

```diff
@@ -222,17 +227,10 @@ const redis = Object.assign({
     },
     reset() {
         return rejectIfNotConnected((resolve, reject) => {
-            const prefix = redisClient.options.prefix;
-
-            redisClient.keys(`${prefix}*`, (err, keys) => {
-                if (err) {
-                    reject(err);
-                } else {
-                    const keysWithoutPrefix = keys.map(slice(prefix.length, Infinity));
-
-                    resolve(Promise.all(keysWithoutPrefix.map(key => this.del(key))));
-                }
-            });
+            this.scan()
+                .then(keys => this.del(keys))
```

Comment:
You can pass just `this.del`

Reply:
passing just `this.del` without an explicit binding, even if it works it doesn't seem correct, I prefer to be explicit in this case.